### DISCUSSION TITLE
Return object when cancelled

### DISCUSSION THREAD

**Discussion by inertia666** [Blazored/Modal#341] [Q&A] [answered]
In some cases it would be nice to have the return object when the modal is cancelled. I want to restore the edit object to its original state where I cannot use bind and have custom event handlers.

```cs
 void Cancel()
    {
      //  ModalInstance.CancelAsync();
        ModalInstance.CloseAsync(ModalResult.Ok<MyObjec>(myOriginalObj));
    }
```
    
Now I have to use an object compare to restore the table I have edited instead of just checking for a cancellation. 

Am i doing this wrong though and something is already in place that I have missed?

**Comment by larsk2009** [comment, marked as answer]
Hmm, it might be useful to have a way to return some things when the modal is cancelled. Could you create an issue for this? That is the best way to track this and have a little discussion about it. Thanks!

**Reply by inertia666** [reply]
Yes, done!
